NGitLab/Models/Event.cs

```diff
@@ -45,26 +45,51 @@ public class Event
         [DataMember(Name = "push_data")]
         public PushData PushData { get; set; }
 
+        /// <summary>
+        /// The target is either a gitlab object (like a merge request)
+        /// or a commit object
+        /// </summary>
+        public string ResolvedTargetTitle
+        {
+            get
+            {
+                if (TargetTitle != null)
+                    return $"{TargetType} '{TargetTitle}'";
+
+                if (PushData != null)
+                {
+                    return $"{PushData.RefType} '{PushData.Ref}'";
+                }
+
+                return string.Empty;
+            }
+        }
+
         /// <summary>
         /// Debug display
         /// </summary>
         /// <returns></returns>
         public override string ToString()
         {
-            return $"{AuthorUserName} {Action} {TargetType} {GetAge(CreatedAt)}";
+            return ToString(ProjectId.ToString());
+        }
+
+        public string ToString(string projectName)
+        {
+            return $"{AuthorUserName} {Action} {ResolvedTargetTitle} at {projectName} ({GetAge(CreatedAt)})";
         }
 
         private static string GetAge(DateTime date)
         {
             var age = DateTime.Now.Subtract(date);
```

Comment:
While this is not part of the changes, this call looks suspicious. I think we should use `UtcNow`.